### PR TITLE
task_4_4_grpc: add a new task

### PR DESCRIPTION
**Pull request by NamesMark**
Please see my suggestion for a new task. I've tried to introduce it in such a way that it doesn't disrupt the flow of existing assignments. After the task 4_3, it should be a rather simple drop-in replacement for the RESTful API between the client and the server.

### CODE REVIEW

**Comment by TheBestTvarynka** [comment]
(rendered README.md: https://github.com/NamesMark/rustcamp/blob/20581f47a001d33fa05ab1f64821a33444164343/4_backend/4_4_grpc/README.md)

**Review by TheBestTvarynka — APPROVED** [review]
I like those changes. `gRPC` is a common protocol for client server communication.
I agree that it's reasonable to add this section. Moreover, it will help students to learn how to write a protocol/framework-agnostic code.

File: 4_backend/4_4_grpc/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+For implementing a [gRPC] server in [Rust], there are two main production-ready crates in its ecosystem: [`tonic`] (pure [Rust] implementation, based on [`tokio`]) and [`grpcio`] (wrapper around [gRPC core][311] implementation).
+
+In [gRPC] ecosystem, usually, implementing a [gRPC] client doesn't differ much from implementing a server, since both are auto-generated from the same `.proto` schema. So, for [Rust], the same [`tonic`] and [`grpcio`] crates do the job when it comes to making [gRPC] requests. 
+
+For more familiarity with using [gRPC] in [Rust], read through the following articles:
+- [Official `tonic` crate docs][`tonic`]
+- [Official `grpcio` crate docs][`grpcio`]
+
+
+
```
Comment (TheBestTvarynka) [inline 4_backend/4_4_grpc/README.md:22]: @NamesMark Can you remove these empty lines. As I remember, one empty like is enough.

**Review by TheBestTvarynka — CHANGES_REQUESTED** [review]

File: 4_backend/4_4_grpc/README.md (hunk 1 above)
Comment (NamesMark) [reply 4_backend/4_4_grpc/README.md]: Absolutely! I like it better this way too.

**Review by TheBestTvarynka — APPROVED** [review]

**Pull request merged**